frappe/installer.py

```diff
@@ -123,6 +127,22 @@ def _new_site(
 	print("*** Scheduler is", scheduler_status, "***")
 
 
+def _rollback_site(
+	db_user=None,
+	setup_db=True,
+):
+	if frappe.flags.db_name:
+		if setup_db:
+			from frappe.database import drop_user_and_database
+
+			drop_user_and_database(frappe.flags.db_name, db_user or frappe.flags.db_name)
+
+		if os.path.isdir(frappe.get_site_path()):
+			import shutil
+
+			shutil.rmtree(frappe.get_site_path())
```

Comment:
IMO `CallbackManager` would be better choice here for rolling back only the changes that have happened. It will also setup a pattern for what needs to be done for any rollback-able action.